### PR TITLE
Components editing

### PR DESCRIPTION
**Pull request by meCloudy**
Easier and simpiler system to edit vehicle components in garage.

### CODE REVIEW

File: gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn [hunk 1]
```diff
@@ -515,30 +432,21 @@ hook OnDialogResponse( playerid, dialogid, response, listitem, inputtext[ ] )
...
+				SetPVarFloat( playerid, "component_rot_z", g_vehiclePimpData[ ownerid ] [ vehicleid ] [ E_RZ ] [ i ] );
 
-				if ( ( g_vehiclePimpData[ ownerid ] [ vehicleid ] [ E_OBJECT ] [ i ] = iObject ) ) {
-					GetVehiclePos( g_vehicleData[ ownerid ] [ vehicleid ] [ E_VEHICLE_ID ], X, Y, Z );
-					SetVehiclePos( g_vehicleData[ ownerid ] [ vehicleid ] [ E_VEHICLE_ID ], X, Y, Z + 0.05 );
-
-					EditDynamicObject( playerid, iObject );
- 				}
+				SendClientMessage( playerid, -1, ""COL_GREY"[GARAGE]"COL_WHITE" You have started editing the component." );
+				SetTimerEx( "OnUpdateVehicleComponents", TIMER_UPDATE_RATE, false, "d", playerid );
```
Comment (zeelorenc) [inline gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn]: Is this timer ever destroyed?

File: gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn [hunk 2]
```diff
@@ -667,7 +575,160 @@ thread OnPlayerCreateVehicleComponent( playerid, vehicleid, slotid )
...
+					SetPVarFloat( playerid, "component_pos_x", GetPVarFloat( playerid, "component_pos_x" ) + VALUE );
+
+			case COMPONENT_EDIT_TYPE_Y:
+					SetPVarFloat( playerid, "component_pos_y", GetPVarFloat( playerid, "component_pos_y" ) + VALUE );
+
+			case COMPONENT_EDIT_TYPE_Z:
+					SetPVarFloat( playerid, "component_pos_z", GetPVarFloat( playerid, "component_pos_z" ) + VALUE );
+
+			case COMPONENT_EDIT_TYPE_RX:
+					SetPVarFloat( playerid, "component_rot_x", GetPVarFloat( playerid, "component_rot_x" ) + ( ( VALUE * 360) / 100 ) * 10 );
```
Comment (zeelorenc) [inline gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn]: fix spaces ( VALUE * 360 )

File: gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn [hunk 3]
```diff
@@ -667,7 +575,160 @@ thread OnPlayerCreateVehicleComponent( playerid, vehicleid, slotid )
...
+	new
+		ownerid = INVALID_PLAYER_ID,
+		vehicleid = GetPlayerVehicleID( playerid ),
+		slotid = GetPVarInt( playerid, "components_editing" ),
+		v = getVehicleSlotFromID( vehicleid, ownerid )
+	;
+	if ( v == -1 || playerid != ownerid || ! g_vehiclePimpData[ ownerid ] [ v ] [ E_CREATED ] [ slotid ] || g_vehiclePimpData[ ownerid ] [ v ] [ E_DISABLED ] [ slotid ] )
+		return SendServerMessage( playerid, "Vehicle component editing was cancelled." ), EndPlayerEditComponent( playerid );
+
+	ShowPlayerHelpDialog( playerid, 60, "~g~/px - ~w~X axis~n~\
```
Comment (zeelorenc) [inline gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn]: This help dialog will only show for "60" milliseconds

It should be 0. That will make it show forever. 

`HidePlayerHelpDialog` or something I made will hide it (put it when they end editing)

File: gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn (hunk 1 above)
Comment (zeelorenc) [reply gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn]:
```cpp
// top of script module
static stock p_EditComponentTimer[ MAX_PLAYERS ] = { -1, ... };

// line 449
KillTimer( p_EditComponentTimer[ playerid ] ); 
p_EditComponentTimer[ playerid ] = SetTimerEx( ..... );

// when you disconnect, close editor, etc... 
KillTimer( p_EditComponentTimer[ playerid ] ); 
p_EditComponentTimer[ playerid ] = -1;
```

Comment (meCloudy) [reply gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn]: > Is this timer ever destroyed?

yes ofc, its not repeating, it just starts every loop on UpdateComponents callback, and when I return a value (to exit the callback) the timer is killed. Easy?

File: gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn (hunk 3 above)
Comment (meCloudy) [reply gamemodes/irresistible/cnr/features/vehicles/vmodifications.pwn]: Dude, the way it works is that, I call the function with a timer at the beginning (SetTimerEx), then the function is called, if there are no errors and everything is cool, the component moves according to what the user presses, and then the function is called again with a timer to repeat.. during repeating, there are if statements in the function to make sure the player is in valid vehicle and valid component etc, if fails to pass the "if statements", it returns 1 so the timer won't be reached (the timer wont run anymore) and the callback wont be called anymore as well.

**Pull request merged**